packages/babel-plugin/src/__tests__/css-builder.test.tsx

```diff
@@ -49,4 +49,51 @@ describe('css builder', () => {
       "
     `);
   });
+
+  it('Calculates a negative varable seperatly from a positive variable of the same value', () => {
+    const actual = transform(`
+      import { styled } from '@compiled/react';      
+      const size = () => 8
+      const gridSize = size();
+
+      const LayoutRight = styled.aside\`
+        margin-right: -\${gridSize * 5}px;
+        margin-left: \${gridSize * 5}px;
+      \`;
+
+      <LayoutRight>Layout Right</LayoutRight>;
+    `);
+
+    expect(actual).toIncludeMultiple([
+      'margin-right:var(--_1cakqv5)',
+      'margin-left:var(--_1l3fmvo)',
+      '"--_1l3fmvo": ix(gridSize * 5, "px")',
+      '"--_1cakqv5": ix(-gridSize * 5, "px"',
+    ]);
+
+    expect(actual).toMatchInlineSnapshot(`
```

Comment:
Snapshot is probably necessary, as you've covered related output assertion above.

Reply:
Updated